ns-provisioner-samples/tanzu-supply-chain/oci-store-credentials.yaml

```diff
@@ -0,0 +1,19 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:json", "json")
+#@ load("@ytt:base64", "base64")
+#@ def generate_dockerconfigjson():
+#@   dcj = {"auths": {}}
+#@   up = {"username": data.values.imported.tanzusupplychain.ocistore.username, "password": data.values.imported.tanzusupplychain.ocistore.password}
+#@   dcj["auths"][data.values.imported.tanzusupplychain.ocistore.server] = up
+#@   return base64.encode(json.encode(dcj))
+#@ end
+---
+apiVersion: v1
+data:
+  .dockerconfigjson: #@ generate_dockerconfigjson()
+kind: Secret
+metadata:
+  name: oci-store-credentials
+  annotations:
+    tekton.dev/docker-0: #@ data.values.imported.tanzusupplychain.ocistore.server + "/" + data.values.imported.tanzusupplychain.ocistore.repository
```

Comment:
Is this annotation being used anywhere? If the data values has just the username password needed for the oci-store-cred and not for oci-store will this setup work?

Reply:
Yes. used by Tekton in the internal workings of the Supply chains